src/components/common/TmSelectNetwork.vue

```diff
@@ -43,32 +44,30 @@ export default {
       } else {
         return `/create`
       }
-    },
-    sortedNetworks() {
-      // sorts networks setting showing the current network first, mainnets at the top and the default one the first
-      if (this.networks.length > 0) {
-        return (
-          [
-            // current network first
-            this.networks.find(({ id }) => id === this.networkId),
-            ...this.networks
-              // ignore the current network in the rest of the list as already showing on the top
-              .filter(({ id }) => id !== this.networkId)
-              // show all mainnets next
-              .sort((a, b) => {
-                return a.testnet - b.testnet
-              })
-              // show the default network on the top
-              .sort((a, b) => {
-                return b.default - a.default
-              })
-          ]
-            // filter out undefineds (happens if this.networkId is not set like in the extension)
-            .filter(x => !!x)
-        )
-      } else {
-        return []
-      }
+    }
+  },
+  mounted: async function() {
```

Comment:
what was the problem and why did you choose this imperative approach?

Reply:
The problem was that the order of the networks was changed after selecting the network. What is the imperative approach?

Reply:
the problem was that networkId changed and therefor the list changed. Why not just remember the networkId on start instead of changing the whole computed (reactive) selectedNetwork list?

Reply:
but not super important

Reply:
But this would be a strange solution for a computed property, don't you think so? Computed property depends on its variables. Moreover is will cause a warning about side effects of mounted property. And the network list could be changed during this step. Based on those thoughts i decided to move in to mounted, as we need only the initial value of this list.

Reply:
exactly. and with a computed property we leverage the asynchronicity of the network request without explicitly waiting for the network request to have passed. we would fix one of the inputs of the computed property `networkId` by setting it once. that's all you wanted to fix.